lib/de.ml

```diff
@@ -3263,24 +3248,28 @@ module Def = struct
       ; unsafe_set_uint8 os.o (os.o_pos + 1) ((v lsr 8) land 0xff)
       ; os.o_pos <- os.o_pos + 2
 
-    (* clecat: Should be rewritten with uint32 writes *)
-    let memcpy os len =
-      let i = ref 0 in
-      while !i < len do
-        let v = unsafe_get_uint8 os.i (os.i_pos + !i) in
-        unsafe_set_uint8 os.o os.o_pos v
-        ; incr i
-        ; os.o_pos <- os.o_pos + 1
+    let memcpy src ~src_off dst ~dst_off ~len =
+      let len0 = len land 3 in
+      let len1 = len asr 2 in
+      for i = 0 to len1 - 1 do
+        let i = i * 4 in
+        let v = unsafe_get_uint32 src (src_off + i) in
+        unsafe_set_uint32 dst (dst_off + i) v
       done
-      ; os.i_pos <- os.i_pos + !i
+      ; for i = 0 to len0 - 1 do
+          let i = (len1 * 4) + i in
+          let v = unsafe_get_uint8 src (src_off + i) in
+          unsafe_set_uint8 dst (dst_off + i) v
+        done
```

Comment:
Can you extract this part to another PR? Just to be sure to separate the PR which fixes the bug and a PR which improve performances.

Reply:
Of course